### PR TITLE
Add support for impersonation in GCP hooks

### PR DESCRIPTION
**Pull request by olchas**
---
Make sure to mark the boxes below before creating PR: [x]

- [ ] Description above provides context of the change
- [ ] Unit tests coverage for changes (not needed for documentation changes)
- [ ] Target Github ISSUE in description if exists
- [ ] Commits follow "[How to write a good git commit message](http://chris.beams.io/posts/git-commit/)"
- [ ] Relevant documentation is updated including usage instructions.
- [ ] I will engage committers as explained in [Contribution Workflow Example](https://github.com/apache/airflow/blob/master/CONTRIBUTING.rst#contribution-workflow-example).

---
In case of fundamental code change, Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvements+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in [UPDATING.md](https://github.com/apache/airflow/blob/master/UPDATING.md).
Read the [Pull Request Guidelines](https://github.com/apache/airflow/blob/master/CONTRIBUTING.rst#pull-request-guidelines) for more information.

### CODE REVIEW

**Comment by olchas** [comment]
@mik-laj, @turbaszek, I added tests for new features - could you take a look at them? If they look fine, I will rebase this branch on apache master and create pull request to the original repo.

File: tests/providers/google/common/hooks/test_base_google.py [hunk 1]
```diff
@@ -632,6 +643,63 @@ def test_authorize_assert_http_timeout_is_present(self, mock_get_credentials):
...
         self.assertNotEqual(http_authorized.timeout, None)
 
+    @mock.patch(MODULE_NAME + '.get_credentials_and_project_id')
+    def test_get_credentials_and_project_id_with_impersonation_chain_as_string(
+        self,
+        mock_get_creds_and_proj_id
+    ):
+        mock_credentials = mock.MagicMock()
+        mock_get_creds_and_proj_id.return_value = (mock_credentials, "PROJECT_ID")
+        self.instance.impersonation_chain = "ACCOUNT_1"
```
Comment (turbaszek) [inline tests/providers/google/common/hooks/test_base_google.py:653]: Does this create a global side effect? Or the instance is created in SetUp?

File: tests/providers/google/common/hooks/test_base_google.py [hunk 2]
```diff
@@ -632,6 +643,63 @@ def test_authorize_assert_http_timeout_is_present(self, mock_get_credentials):
         http_authorized = self.instance._authorize().http
         self.assertNotEqual(http_authorized.timeout, None)
 
+    @mock.patch(MODULE_NAME + '.get_credentials_and_project_id')
+    def test_get_credentials_and_project_id_with_impersonation_chain_as_string(
+        self,
+        mock_get_creds_and_proj_id
+    ):
+        mock_credentials = mock.MagicMock()
+        mock_get_creds_and_proj_id.return_value = (mock_credentials, "PROJECT_ID")
```
Comment (turbaszek) [inline tests/providers/google/common/hooks/test_base_google.py:652]:
```suggestion
        PROJECT_ID = "PROJECT_ID"
        mock_get_creds_and_proj_id.return_value = (mock_credentials, PROJECT_ID)
```
I would suggest using a global constant for that

File: tests/providers/google/common/hooks/test_base_google.py [hunk 3]
```diff
@@ -632,6 +643,63 @@ def test_authorize_assert_http_timeout_is_present(self, mock_get_credentials):
...
+        self.assertEqual((mock_credentials, "PROJECT_ID"), result)
+
+    @mock.patch(MODULE_NAME + '.get_credentials_and_project_id')
+    def test_get_credentials_and_project_id_with_single_account_in_impersonation_chain(
+        self,
+        mock_get_creds_and_proj_id
+    ):
+        mock_credentials = mock.MagicMock()
+        mock_get_creds_and_proj_id.return_value = (mock_credentials, "PROJECT_ID")
+        self.instance.impersonation_chain = ["ACCOUNT_1"]
```
Comment (turbaszek) [inline tests/providers/google/common/hooks/test_base_google.py:672]: Should we use parametrize for this value? Both tests seem very similar

Comment (olchas) [reply tests/providers/google/common/hooks/test_base_google.py:672]: Yes, I think we can parametrize even all three new tests from this class. I will change that.

File: tests/providers/google/common/hooks/test_base_google.py (hunk 2 above)
Comment (olchas) [reply tests/providers/google/common/hooks/test_base_google.py:652]: I was thinking about a global constant, but other tests present in this module were also using hardcoded strings. Should I change them to use the global constant as well? Or just the new tests?

Comment (turbaszek) [reply tests/providers/google/common/hooks/test_base_google.py:652]: New tests will be ok, we can fix others next time

File: tests/providers/google/common/hooks/test_base_google.py (hunk 1 above)
Comment (olchas) [reply tests/providers/google/common/hooks/test_base_google.py:653]: `self.instance` is indeed created in `setUp`, so as far as I know, there should be no side effect.

**Pull request closed without merging**